src/Tasks/CheckConfigurationAndPlatform.cs

```diff
@@ -0,0 +1,98 @@
+﻿using Microsoft.Build.Shared;
+using System;
+
+namespace Microsoft.Build.Tasks
+{
+    /// <summary>
+    /// Verifies some critical properties like OutputPath are set.  This runs as part of the _CheckForInvalidConfigurationAndPlatform target.
+    /// </summary>
+    public class CheckConfigurationAndPlatform : TaskExtension
```

Comment:
Rather than make a specialized task, why not make general localized Error and Warning tasks (maybe subclass the existing ones if that fits) that take in a resource name?

Reply:
I started with that but couldn’t figure out a good way to specify the assembly and resource. You would need one Error task per assembly that contained resources right?  And if you specify an assembly in XML, it would need to be a path. And then the way you call it with format args would be interesting. Any ideas?

Reply:
Maybe just always have it be a resource in the Tasks resources, and just pass in a name? It'd only work for us, but it'd be a step in the right direction, I think.

Passing in things for the format string is definitely interesting. Maybe pass in an item and have metadata like Field1 Field2? That's not exactly elegant :(

Reply:
See https://github.com/dotnet/sdk/blob/master/src/Tasks/Common/MessageBase.cs and https://github.com/dotnet/sdk/blob/master/src/Tasks/Common/NETSdkError.cs for how we handle this in dotnet/sdk